localstack/services/apigateway/provider.py

```diff
@@ -207,23 +201,30 @@ def put_rest_api(self, context: RequestContext, request: PutRestApiRequest) -> R
 
         openapi_spec = parse_json_or_yaml(to_str(body_data))
         rest_api = import_api_from_openapi_spec(
-            rest_api, openapi_spec, context.request.values.to_dict()
+            rest_api,
+            openapi_spec,
+            context.request.values.to_dict(),
+            account_id=context.account_id,
+            region=context.region,
         )
 
         response = rest_api.to_dict()
         remove_empty_attributes_from_rest_api(response)
+        store = get_apigateway_store(account_id=context.account_id, region=context.region)
+        store.rest_apis[request["restApiId"]].rest_api = response
```

Comment:
could we wrap this into `store.add_rest_api(api_id, response)` same for `delete` ?